core/modules/dashboard/dashboard.module

```diff
@@ -0,0 +1,444 @@
+    $items = $cache->data;
+  }
+  else {
+    // User Overview.
+    if (!$settings['user'] || $settings['user'] == 1) {
+      $items['all'] = db_query("SELECT count(*) FROM {users}")->fetchField();
+      $items['active'] = db_query("SELECT count(*) FROM {users} WHERE status = 1 AND login <> 0")->fetchField();
+      $items['blocked'] = db_query("SELECT count(*) FROM {users} WHERE status = 0")->fetchField();
+      // Account for UID 0. Faster than adding a where clause.
+      $items['all']--;
+      $items['blocked']--;
+    }
+    // Roles Overview.
+    $roles = user_roles(TRUE);
+    $items['roles'] = array();
+    foreach ($roles as $rid => $role) {
+      if (($rid != 2) && (array_key_exists($rid, $settings['roles']) && $settings['roles'][$rid])) {
+        $user_role_count = db_query("SELECT count(*) FROM {users} u INNER JOIN {users_roles} r on u.uid = r.uid WHERE r.role = :rid", array(':rid' => $rid))->fetchField();
+        $items['roles'][$rid] = array();
+        $items['roles'][$rid]['name'] = $role;
+        $items['roles'][$rid]['count'] = $user_role_count;
+      }
+    }
+
+    cache_set('dashboard_user_overview', $items, 'cache');
+  }
+
+  return $items;
+}
+/**
+ * Adds user settings to the block settings form.
+ *
+ * @param $form
+ *   Block settings form.
+ * @param $settings
+ *   Block settings data.
+ *
+ */
+function dashboard_add_user_block_settings(&$form, $settings = array()) {
+  $user_options = array(
+    0 => t('no'),
+    1 => t('Include the total number of user accounts, including active and blocked.'),
+  );
+  $form['user'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Include the total number of user accounts, including active and blocked.'),
+    '#options' => $user_options,
+    '#default_value' => ($form_state['op'] == 'add') ? TRUE : $settings['user'],
+  );
+  $roles = user_roles(TRUE);
+  $roles_options = array();
+  $roles_defaults = array();
+  foreach ($roles as $rid => $role) {
+    if ($rid != 2) {
+      $roles_options[$rid] = $role;
+      $roles_defaults[] = $rid;
+    }
+  }
+  if (!empty($roles_options)) {
+    $form['roles'] = array(
+      '#type' => 'checkboxes',
+      '#title' => t('Include the number of user accounts with the following roles:'),
+      '#options' => $roles_options,
+      '#default_value' => ($settings['roles']) ? $settings['roles'] : $roles_defaults,
+    );
+  }
+}
+
+/**
+ * menu callback for settings page
+ */
+function dashboard_admin_settings() {
+   $config = config('dashboard.settings');
+  // Redirect on log in.
+  $form['#config'] = 'dashboard.settings';
+  $form['dashboard_login_redirect'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Redirect administrators to Dashboard after log in'),
+    '#default_value' => $config->get('dashboard_login_redirect'),
+    '#description' => t('Everyone with the !permission permission will be automatically redirected to the Dashboard when they log in..', array(
```

Comment:
```suggestion
    '#description' => t('User accounts with the !permission permission will be automatically redirected to the Dashboard when they log in.', array(
```

Double-period at the end. And perhaps use "User accounts" instead of "Everyone"?

Reply:
fixed by @quicksketch